azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/CallComposite.java

```diff
@@ -428,15 +728,70 @@ private void showUI(final Context context,
 
         CallCompositeInstanceManager.putCallComposite(instanceId, this);
 
+        initializeCallAgent();
+
         final Intent intent = new Intent(context, activityClass);
         intent.putExtra(CallCompositeActivity.KEY_INSTANCE_ID, instanceId);
-        if (isTest) {
+        if (isTest || isAcceptIncomingCall) {
             intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         }
 
         context.startActivity(intent);
     }
 
+    private void handlePushNotification(final Context context,
```

Comment:
If possible, handle 1:n in another class, and delegate out, for the following:

1. You can then supply mock handler, and drop the isTest
2. CallComposite is basically a header to a user. They'll use it to quickly reference our API, so we should abstract implementations away from it.

Another things to consider is using the type system on CallCompositeRemoteOptions.

You can potentially make an internal OneToNCallCompositeRemoteOptions : CallCompositeRemoteOptions

Then you can use the type system in the natural flows to distinguish of it's a RemoteOptions created via push, or RemoteOptions created by a user (with public class).

Just a thought though. The less API/Implementation in CallComposite the better.

Reply:
As per discussion we will fix for GA.